### DISCUSSION TITLE
[solved] Get z-coordinate of box in HFSS 3D modeler

### DISCUSSION THREAD

**Discussion by fwlfHS** [ansys/pyaedt#1341] [Q&A] [answered]
Hi everyone,
maybe someone may help me out with a nice solution.
I have a box created in HFSS named "Line" and would like to extract the z-coordinate (lower side of the box)

In the past I used the old API to do this:
```
list_objects = [oEditor.GetObjectName(i) for i in range(oEditor.GetNumObjects())]
name = 'Line'
if name in list_objects:
   line_z = oEditor.GetPropertyValue('Geometry3DCmdTab', name + ':CreateBox:1', 'Position').split(',')[2]
```

EDIT: 
**Solved** by fetching the bottom face center coordinates and use it as position:
```
from pyaedt import Hfss
hfss = Hfss()
if hfss.modeler.does_object_exist('Line'):
    line = hfss.modeler.get_object_from_name('Line')
    z_pos = line.bottom_face_z.center[2] 
```

**Comment by maxcapodi78** [comment, marked as answer]
@fwlfHS your approach is correct. you have a lot of properties for the objects including edges and vertices for which you can get midpoints and position.
